intermol/gromacs/gromacs_parser.py

```diff
@@ -213,9 +194,6 @@ def canonical_dihedral(self, params, dihedral, direction='into'):
             if dihedral == ProperPeriodicDihedralType:
                 convertfunc = convert_dihedral_from_proper_to_trig
                 converted_dihedral = TrigDihedralType
-            elif dihedral == ProperPeriodicDihedral:
-                convertfunc = convert_dihedral_from_proper_to_trig
-                converted_dihedral = TrigDihedral
```

Comment:
Everything passed into this function during reading should already be a `XYZType`. Let me know if you find a code path that does not do this!